core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -123,7 +123,7 @@ private[spark] class TaskSetManager(
   // state until all tasks have finished running; we keep TaskSetManagers that are in the zombie
   // state in order to continue to track and account for the running tasks.
   // TODO: We should kill any running task attempts when the task set manager becomes a zombie.
-  private[scheduler] var isZombie = false
+  @volatile private[scheduler] var isZombie = false
```

Comment:
I don't think this is necessary.  You still only touch `isZombie` with a lock on the TaskSchedulerImpl (*everything* in the TSM requires a lock on the TaskSchedulerImpl).  Even before, there could be multiple task-result-getter threads interacting w/ this.

Reply:
ah good catch!